tests/RobotFramework/tests/cumulocity/software_management/software.robot

```diff
@@ -69,6 +69,44 @@ Manual software_update operation request
     ...    expected_status=failed
     ...    c8y_fragment=c8y_SoftwareUpdate
 
+Operation log uploaded automatically with auto_log_upload setting as on-failure
+    Execute Command    tedge config set c8y.operations.auto_log_upload on-failure
+    Restart Service    tedge-mapper-c8y
+
+    # Validate that the operation log is NOT uploaded for a successful operation
+    ${OPERATION}=    Install Software        c8y-remote-access-plugin
```

Comment:
I would suggest Parametrizing test data, it can enhance test coverage and maintainability by allowing the same test case to be executed with different inputs. for example:
```
*** Variables *** 
${SUCCESS_PACKAGE}    c8y-remote-access-plugin 
${FAILED_PACKAGE}     non-existent-package

*** Test Cases ***
Operation log uploaded automatically with default auto_log_upload setting as onFailure
    ${OPERATION}=    Install Software        ${SUCCESS_PACKAGE}
    ...

Operation log uploaded automatically with default auto_log_upload setting as always
    ${OPERATION}=    Install Software        ${SUCCESS_PACKAGE}
    ...

Operation log uploaded automatically with default auto_log_upload setting as never
    ${OPERATION}=    Install Software        ${SUCCESS_PACKAGE}
```